# Path of Exile game stats data eda

dates: 2022.04.21 ~ 

This file contains an analysis processing of the game 'Path of Exile' data.

###### Origins of datas from here: 
poe_stats : https://www.kaggle.com/datasets/gagazet/path-of-exile-league-statistic

Data contains stats of 59000 players, from 4th August of 2017 and before now.




# Dataset exploring

## data and libraries import

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

poe_stats = pd.read_csv('poe_stats.csv')

In [2]:
poe_stats

rank   dead  online                    name  level        class  \
0          1  False   False      Tzn_NecroIsFineNow    100  Necromancer   
1          1  False   False        RaizNeverFirstQT    100  Necromancer   
2          1  False   False  GucciStreamerAdvantage    100  Necromancer   
3          1  False   False             ChiroxPrime    100       Slayer   
4          2  False   False     Cool_NecroIsFineNow    100      Deadeye   
...      ...    ...     ...                     ...    ...          ...   
59771  14999   True   False       ПроклятьеРекласта     89  Necromancer   
59772  15000  False   False       IshibashiSummoner     94  Necromancer   
59773  15000  False   False                 BLively     73       Slayer   
59774  15000  False   False                vawddvaw     89    Gladiator   
59775  15000  False   False                 Reselin     53  Necromancer   

                                                      id  experience  \
0      3dcddd59f5088893f734f39686350990dae168cc4f4b00...  4250334444   
1      8f3216db5ac9106c287a834731aafc83c387138f28fc45...  4250334444   
2      c6ec2dae3855c551e0597c06ef2da06fbb5512487ded7f...  4250334444   
3      c861372da792be0b22c45bf437ccd58437c52e9455e0bf...  4250334444   
4      24ae924ceed7989ef3d3d6772612832bb467a6094352c7...  4250334444   
...                                                  ...         ...   
59771  d33b4f6e08c10e365765f9a36a8f36d561fd1d86f10ee1...  1786695653   
59772  5764cfa387e0a87a4bebc1a3c5017e92de8bbb06445c47...  2720770853   
59773  9ac75ab75a47cee8a9dfb0a31912df8909720a8b20f9f1...   495034029   
59774  cf02dfc0c90b2df9c7ac76bbedd91e93c2a8a2ca629c15...  1786638972   
59775  f7ffda5ca2490546344d32930693f8299930ce9451df4a...    77909816   

            account  challenges        twitch              ladder  
0            TheTzn          40        thetzn           Harbinger  
1            RaizQT          22        raizqt    SSF Harbinger HC  
2       GucciPradas          35   guccipradas  Hardcore Harbinger  
3          Chiroxun          26           NaN       SSF Harbinger  
4          cooltail          22           NaN           Harbinger  
...             ...         ...           ...                 ...  
59771  Romanchik002          18  romanchik002  Hardcore Harbinger  
59772  Ishibashisan          33           NaN           Harbinger  
59773        ruby01          19           NaN       SSF Harbinger  
59774        Lecarn          24        lecarn  Hardcore Harbinger  
59775      woworpoe          19           NaN    SSF Harbinger HC  

[59776 rows x 12 columns]

In [13]:
df.sample(5)

rank   dead  online                 name  level        class  \
33344   8359   True   False           SSF_Bakuro     70  Necromancer   
49186  12340   True   False     maythefrostbewin     89   Hierophant   
28432   7127  False   False             HaciHuso     95       Slayer   
51546  12931   True   False  WoschmaonaufdaGrodn     89    Berserker   
57545  14440  False   False          DJMurdaFace     74    Trickster   

                                                      id  experience  \
33344  8033de294406fe0a2faea6a0cfa5fb2a5e75507d9c40d9...   363906163   
49186  22bfcab1ddf3eb72232410a624645e7c18c9a4ed0fcd51...  1899285411   
28432  52b500a6b08813dc8bd0c5e7932fbd3615c88643fd82ac...  2949772118   
51546  050b38bee36353547377ca9e04a14f73fa3b7b1676243d...  1868647029   
57545  941161ff8b3cfde2d3ecb0bb9eadccad16a6bd3c87ecc4...   521654616   

            account  challenges           twitch              ladder  
33344    ZenMasters          14              NaN    SSF Harbinger HC  
49186       mad_fan          33              NaN  Hardcore Harbinger  
28432   Sentencedfb          37  hammerofancient           Harbinger  
51546        Shaarq          24    shaarq_twitch  Hardcore Harbinger  
57545  striderremix          20              NaN       SSF Harbinger

In [6]:
df.shape

(59776, 12)

## data describes

In [3]:
df = poe_stats
df.describe()

rank         level    experience    challenges
count  59776.000000  59776.000000  5.977600e+04  59776.000000
mean    7495.369379     85.422578  1.825786e+09     25.759569
std     4330.179898     11.288017  1.094748e+09      8.440888
min        1.000000     53.000000  7.790982e+07      4.000000
25%     3745.000000     78.000000  7.774697e+08     19.000000
50%     7492.000000     90.000000  1.934010e+09     25.000000
75%    11247.000000     95.000000  2.876117e+09     33.000000
max    15000.000000    100.000000  4.250334e+09     40.000000

In [4]:
df.describe(include='object')

name      class                                                 id  \
count    59776      59776                                              59776   
unique   58286         26                                              59776   
top     _Flic_  Berserker  3dcddd59f5088893f734f39686350990dae168cc4f4b00...   
freq        14       8711                                                  1   

        account            twitch         ladder  
count     59776             16300          59776  
unique    37123              9613              4  
top     chuz101  thespunkylobster  SSF Harbinger  
freq         30                18          14981

In [8]:
df.columns

Index(['rank', 'dead', 'online', 'name', 'level', 'class', 'id', 'experience',
       'account', 'challenges', 'twitch', 'ladder'],
      dtype='object')

In [10]:
df.nunique()

rank          15000
dead              2
online            2
name          58286
level            48
class            26
id            59776
experience    54124
account       37123
challenges       37
twitch         9613
ladder            4
dtype: int64

In [15]:
df.isnull().sum()

rank              0
dead              0
online            0
name              0
level             0
class             0
id                0
experience        0
account           0
challenges        0
twitch        43476
ladder            0
dtype: int64

### describe results

데이터 총 개수 59776 개

컬럼명: ['rank', 'dead', 'online', 'name', 'level', 'class', 'id', 'experience', 'account', 'challenges', 'twitch', 'ladder']

랭크는 총 1만 5천개, 죽음과 온라인 여부는 True, False로 이루어져있다.

id가 오직 unique 하며, 한 아이디당 여러 account를 가질 수 있는 것으로 보인다. 그에 따라 일부 name은 겹치는 것을 보인다.

level 과 class는 게임 내 레벨과 직업으로 이루어져 있다.

experience 는 게임 내 경험치를 의미하는 것일 것이며,

twitch는 트위치 계정이 적혀있는 컬럼으로 보인다. 

##### ladder의 경우 

고유 값이 4개로, 원본 데이터 문서를 찾아본 결과 

```
One file with 12 data sections. One league - Harbinger, but 4 different types of divisions:
Harbinger
Hardcore Harbinger
SSF Harbinger
SSF Harbinger HC
Each division has own ladder with leaders
```
라고 설정되어있다. 

https://pathofexile.fandom.com/wiki/Harbinger_league#Challenges_and_Rewards 와 
https://pathofexile.fandom.com/wiki/Solo_Self-Found 를 참조한 결과 
> Harbinger라는 이름의 시즌 리그의 일종으로 보이며 HC는 Hardcore의 약어로 보인다. Hardcore Harbinger에서 죽은 경우 일반 리그로 옮겨진다는 것과, SSF 가 Solo self-found 로 솔로플레이 모드(파티 거부)의 일종인 점 등을 미루어 보아 난이도는 Harbinger < Hardcore Harbinger < SSF Harbinger < SSF Harbinger HC 가 아닐까 예상된다. 물론 파티와 솔로 플레이 모드는 구분을 해야한다. 클래스 종류 등도 각 게임 모드마다 차이가 날 것이다.


### EDA planning

이에 따라 현 데이터 셋은 이렇게 나누어 볼 수 있을 것이다.

1. rank (rank bins) 
2. level or exp bins
3. class
4. laddar

또한 위의 내용을 laddar 로 나누어 다시 나누어 볼 수 있을 것이다.